docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -238,17 +238,148 @@ the strategy employed by the Example HTTP connector.
 .. _connector-record-set-provider:
 
 ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+--------------------------
+
+Given a split and a list of columns, the record set provider is responsible for
+delivering data to the Trino execution engine. It creates a ``RecordSet``, which
+in turn creates a ``RecordCursor`` that's used by Trino to read the column
+values for each row.
+
+The provided record set must only include requested columns in the order
+matching the list of column handles passed to the
+``ConnectorRecordSetProvider.getRecordSet()`` method. It must return all the
+rows contained in the "virtual table" represented by the TableHandle associated
+with the TableScan operation. It shouldn't return more or fewer rows.
```

Comment:
Replace "It" with "This set" or similar

Reply:
"should return the exact same number of rows" is clearer to me than "shouldn't return more or fewer"

Reply:
I dropped the last sentence as in this form it would just duplicate the previous one.